src/core/solvers/ipopt/multiple-shooting-nlp.cpp

```diff
@@ -0,0 +1,589 @@
+#include "coin-or/IpIpoptApplication.hpp"
+
+#include "crocoddyl/core/solvers/ipopt/multiple-shooting-nlp.hpp"
+
+#include <cassert>
+#include <cmath>
+
+#include <iostream>
+
+#ifdef __GNUC__
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+#endif
+
+MultipleShootingNlp::MultipleShootingNlp(const boost::shared_ptr<crocoddyl::ShootingProblem> &problem)
+    : problem_(problem),
+      state_(problem_->get_runningModels()[0]->get_state()),
+      nx_(problem_->get_nx()),
+      ndx_(problem_->get_ndx()),
+      nu_(problem_->get_nu_max()),
```

Comment:
Let's do not allocate the maximum nu.
This got deprecated recently (see `SolverDDP` in devel branch for more details).

Reply:
This implies that you need to modify a few lines of code, I'll let you to figure out them and later I could make comments on it.